pkg/controllers/managementuser/rbac/roletemplates/register.go

```diff
@@ -9,7 +9,9 @@ import (
 func Register(ctx context.Context, workload *config.UserContext) {
 	management := workload.Management.WithAgent("rbac-role-templates")
 
-	management.Management.ClusterRoleTemplateBindings("").AddLifecycle(ctx, "cluster-crtb-handler", newCRTBLifecycle(workload))
+	c := newCRTBHandler(workload)
+	management.Wrangler.Mgmt.ClusterRoleTemplateBinding().OnChange(ctx, "cluster-crtb-change-handler", c.OnChange)
+	management.Wrangler.Mgmt.ClusterRoleTemplateBinding().OnRemove(ctx, "cluster-crtb-remove-handler", c.OnRemove)
```

Comment:
Sorry we may have talked about this but my memories are vague: why should we move from Lifecycle to separate OnChange + `OnRemove`? Is there any difference in behavior?
Beware that existing objects may already have a finalizer for the previous `cluster-crtb-handler` Lifecycle handler, which would remain unhandled with this change. We need to either pro-actively remove it (in `OnChange`) or just make this use the older name.